DegreeProject/Assets/Scripts/Player/Player.cs

```diff
@@ -242,6 +256,16 @@ private void OnAttackInputDown()
         if (!AnimationIsPlaying(StringData.attack))
         {
             animator.SetTrigger(StringData.attack);
+
+            Collider2D[] hitByAttack = Physics2D.OverlapCircleAll(attackPoint.position, attackRange, LayerMask.GetMask(StringData.enemyLayer));
+
+            foreach (var col in hitByAttack)
+            {
+                if (col.GetComponent<AI>() != null)
+                {
+                    col.GetComponent<AI>().TakeDamage(damage);
```

Comment:
Här kan du göra det lite mer performant genom att antingen spara ner värdet 
```
AI ai = col.GetComponent<AI>();
if(ai != null)
{ ai.TakeDamage(damage); }
```

Alt. använda nya fancy C# syntax sugar ([Null conditional operator](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/member-access-operators#null-conditional-operators--and-))
`col.GetComponent<AI>()?.TakeDamage(damage);`